### PR TITLE
[SPARK-47304][SQL][TESTS] Distribute tests from `DataFrameSuite` to more specific suites

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to move some tests from `DataFrameSuite` out, and distribute them across more specific tests suites and new one `DataFrameShowSuite`.

### Why are the changes needed?
1. Improve maintainability of `DataFrameSuite`
2. Speed up execution of the test suite. Execution time dropped to < 1min.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By running the modified test suites.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameShowSuite.scala [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+package org.apache.spark.sql
+
+import java.nio.charset.StandardCharsets
+import java.sql.{Date, Timestamp}
+
+import org.apache.spark.sql.functions.rand
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.test.SharedSparkSession
+
+class DataFrameShowSuiteextends extends QueryTest with SharedSparkSession {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameShowSuite.scala:27]:
```suggestion
class DataFrameShowSuite extends QueryTest with SharedSparkSession {
```

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 2]
```diff
@@ -2200,6 +2200,115 @@ class DataFrameAggregateSuite extends QueryTest
...
+        val structDf = largeDecimals.select("a").agg(sum("a"))
+        assertDecimalSumOverflow(structDf, ansiEnabled, Row(null))
+      }
+    }
+  }
+
+  test("SPARK-32761: aggregating multiple distinct CONSTANT columns") {
+     checkAnswer(sql("select count(distinct 2), count(distinct 2,3)"), Row(1, 1))
+  }
 }
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:2313]: just for curiosity, what's the runtime of `DataFrameAggregateSuite`? It has more than 2000 LOC...

Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:2313]: 1 min 5 sec on my laptop.

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameToSchemaSuite.scala [hunk 3]
```diff
@@ -365,4 +367,57 @@ class DataFrameToSchemaSuite extends QueryTest with SharedSparkSession {
       errorClass = "NOT_NULL_CONSTRAINT_VIOLATION.MAP_VALUE",
       parameters = Map("columnPath" -> "`map`"))
   }
+
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameToSchemaSuite.scala]: moved

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @cloud-fan for review.

**Pull request closed without merging**